x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/TimeSyncConfig.java

```diff
@@ -37,28 +38,26 @@
 
     private static ConstructingObjectParser<TimeSyncConfig, Void> createParser(boolean lenient) {
         ConstructingObjectParser<TimeSyncConfig, Void> parser = new ConstructingObjectParser<>(NAME, lenient,
-                args -> {
-                    String field = (String) args[0];
-                    TimeValue delay = args[1] != null ? (TimeValue) args[1] : TimeValue.ZERO;
-
-                    return new TimeSyncConfig(field, delay);
-                    });
-
+            args -> {
+                String field = (String) args[0];
+                TimeValue delay = (TimeValue) args[1];
+                return new TimeSyncConfig(field, delay);
+            });
         parser.declareString(constructorArg(), DataFrameField.FIELD);
         parser.declareField(optionalConstructorArg(),
-                (p, c) -> TimeValue.parseTimeValue(p.textOrNull(), DataFrameField.DELAY.getPreferredName()), DataFrameField.DELAY,
-                ObjectParser.ValueType.STRING_OR_NULL);
-
-                    return parser;
-                }
+            (p, c) -> TimeValue.parseTimeValue(p.textOrNull(), DEFAULT_DELAY, DataFrameField.DELAY.getPreferredName()),
+            DataFrameField.DELAY,
+            ObjectParser.ValueType.STRING_OR_NULL);
```

Comment:
I know you didn't introduce it in this PR, but since you're changing this line of code please can you change `STRING_OR_NULL` to just `STRING` and `p.textOrNull()` two lines higher to `p.text()`.  `TimeValue.parseTimeValue()` will throw NPE if passed `null`, so we're not helping anyone by having the parser accept an explicit `null`.  (Not present at all is fine but doesn't go through the `null` code path.)

Reply:
Definitely!

Reply:
FWIW `null` was handled in the parser before